### ISSUE TITLE
Add "owned" error types

### ISSUE THREAD

**Issue by lucacasonato** [parcel-bundler/parcel-css#133]
Currently many error types are have attached lifetimes. This means they can not be boxed and bubbled  up. It'd be great if errors were fully "owned", and did not have attached lifetimes. I don't think this should be super terrible for perf, as errors are usually very much an edge case.

The bigger question is: how could this be done?

**Comment by lucacasonato** [comment]
It is possible to circumvent the lifetimes, by serialising the error to a string, and then boxing and bubbling that string. This doesn't allow proper error introspection though (I don't know how important this is to people). In Deno we will get away with formatting the error to a string.

**Comment by devongovett** [comment]
Yep the hard part would be when a cssparser::Token is included in the error. We'd need to have an ErrorToken or something that is the same but with String instead of CowRcStr.

**Comment by DanielJoyce** [comment]
Also since the errors can hold references to the things that caused it, this expands the bounds of some inputs in certain functions to &'static. One example is Bundler::new and using it inside a spawned future...
